sdk/lib/core/uri.dart

```diff
@@ -576,38 +635,38 @@ abstract interface class Uri {
 
   /// Whether the URI is absolute.
   ///
-  /// A URI is an absolute URI in the sense of RFC 3986 if it has a scheme
-  /// and no fragment.
+  /// A URI is an absolute URI in the sense of RFC 3986 if it has a [scheme]
+  /// and no [fragment].
   bool get isAbsolute;
 
   /// Whether the URI has a [scheme] component.
   bool get hasScheme => scheme.isNotEmpty;
 
-  /// Whether the URI has an [authority] component.
+  /// Whether the URI has an [authority] (hostname) component.
   bool get hasAuthority;
 
-  /// Whether the URI has an explicit port.
+  /// Whether the URI has an explicit [port].
   ///
   /// If the port number is the default port number
   /// (zero for unrecognized schemes, with http (80) and https (443) being
   /// recognized),
   /// then the port is made implicit and omitted from the URI.
   bool get hasPort;
 
-  /// Whether the URI has a query part.
+  /// Whether the URI has a [query] part.
```

Comment:
I do like the link, but it's *technically* a little misleading.
The `Uri` always has a `[query]` getter. It may or may not have a *query component*. (If not, the `query` getter returns the empty string.)

Keep it for now. It links to useful information.